app/scripts/modules/core/src/pipeline/config/stages/stage.module.js

```diff
@@ -96,20 +98,23 @@ module.exports = angular
         }
       };
 
-      $scope.updateAvailableDependencyStages = function() {
-        var availableDependencyStages = PipelineConfigService.getDependencyCandidateStages(
-          $scope.pipeline,
-          $scope.stage,
-        );
-        $scope.options.dependencies = availableDependencyStages.map(function(stage) {
+      $scope.producesArtifactsChanged = function (artifacts) {
+        $scope.$applyAsync(() => {
+          $scope.stage.expectedArtifacts = artifacts;
+        });
+      };
+
+      $scope.updateAvailableDependencyStages = function () {
+        var availableDependencyStages = PipelineConfigService.getDependencyCandidateStages($scope.pipeline, $scope.stage);
+        $scope.options.dependencies = availableDependencyStages.map(function (stage) {
           return {
             name: stage.name,
             refId: stage.refId,
             available: true,
           };
         });
 
-        $scope.pipeline.stages.forEach(function(stage) {
+        $scope.pipeline.stages.forEach(function (stage) {
```

Comment:
There are a lot of style changes throughout this PR that add noise to the diff, many of which are not consistent with the rest of the Deck codebase.  Please try to avoid mixing large feature modifications alongside lints.

Reply:
As far as I can tell, it only happened in this file. Reverted style changes in 4dff56206.